### ISSUE TITLE
Properties of type date are not recognised on template transclusion

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticMediaWiki#1028]
SMW 2.2.0, MW 1.23.9, PHP 5.4

Contents of "Template:TranscludedStartDate":
`[[Has start date::{{{StartDate|}}}]]`
or 
`{{#set:Has start date={{{StartDate|}}} }}`
where property "Has start date" is of datatype "date".

Contents of page "StartDate"
`{{TranscludedStartDate|StartDate=2015/05/16}}`
results into a value 2015/05/16 of type "page" shown on special "Browse" for property "Has start date". However, no value is shown on page "Property:Has start date."

If done similarly for properties of type e.g. "URL" or "Number" this setup works like charm, i.e. the values are correctly recognised and stored.

**Comment by mwjames** [comment]
Just tried your example for those above examples and when pressing purge both data appear as expected.

**Comment by kghbln** [comment]
Cannot tell what is wrong here. Works for me except for type date. Whatever I do (purging, resaving) I get a page link.

**Comment by mwjames** [comment]
![image](https://cloud.githubusercontent.com/assets/1245473/7667905/b68a5a28-fc1d-11e4-88bb-ea6968953d03.png)

![image](https://cloud.githubusercontent.com/assets/1245473/7667909/cc2a383a-fc1d-11e4-80a8-03df1faba1f5.png)

![image](https://cloud.githubusercontent.com/assets/1245473/7667911/d7376e32-fc1d-11e4-8ca3-d6f9a6cac5f9.png)

![image](https://cloud.githubusercontent.com/assets/1245473/7667913/e0082be6-fc1d-11e4-97a7-e5512576207b.png)

**Comment by kghbln** [comment]
This is very annoying, probably not just for you. I just tried to recreate this problem on a public wiki. There it works too. I already spent an hour on this and the second wiki running on the same codebase and it is just behaving differently. I actually planned 1 minute time for this bloody template.

**Comment by mwjames** [comment]
> This is very annoying, probably not just for you.

If there is an issue then we want grasp it.

>  I actually planned 1 minute time for this bloody template.

I know what you mean.

PS: For example, I just switched to PHP 5.6 earlier today and saw an issue with the opcode cache used so I had to change `opcache.revalidate_freq=10`.

```
[opcache]
zend_extension="D:\xampp\php\ext\php_opcache.dll"
opcache.memory_consumption=128
opcache.interned_strings_buffer=8
opcache.max_accelerated_files=4000
opcache.revalidate_freq=10
opcache.fast_shutdown=1
opcache.enable_cli=1

[apcu]
extension="D:\xampp\php\ext\php_apcu.dll"
apc.enabled=1
apc.shm_size=32M
apc.ttl=7200
apc.enable_cli=1
apc.serializer=php
```

**Comment by kghbln** [comment]
I got it to work now, but I have no idea what was wrong - just a theory. Basically I recreated the original template step by step on a fresh template and tested each step if it was working which it did. Then I moved the new identical template over the old one. That's it.

The only theory I can come up with is that at one stage I must have copied the property to another spot within the template (did so twice) and accidentally added a hidden `&shy;` to the property name while doing so. Dunno how I could have done this, but this is the only thing I can think of which would give me the same property name but a type "page" data value. An indicator that my verify this theory is that I currently indeed have two properties of the same name, one with the type "date" assigned and one with the type "page" assigned.
